### DISCUSSION TITLE
Lsp cmd_cwd does not match buffer cwd

### DISCUSSION THREAD

**Discussion by jokajak** [LazyVim/LazyVim#235] [Q&A] [answered]
I'm having an issue with my workspace root detection because the Lsp servers that get attached have their current working directory for running the command set to a path 2 levels higher than my current working directory and I can't figure out how it works to debug.

Using `LspInfo` I've determined that the `root directory` for the client is set to `/path/to/git/work/`. My `pwd` output is `/path/to/git/work/repo.git/develop`

I then used the super hacky `:lua print(vim.pretty_print(vim.lsp.get_active_clients({ bufnr = 0})))` followed by `:Noice` to find that the `cmd_cwd` is set to `/path/to/git/work/`.

I've verified that `require("lspconfig.util").find_git_ancestor("/path/to/git/work/repo.git/develop")` returns `/path/to/git/work/repo.git/develop`. This is the `root_dir` for yamlls which is one of the LSPs I'm using.

Any suggestions on how to continue to diagnose and figure what is going wrong?

**Comment by folke** [comment, marked as answer]
you need fix the root_dir detection for the lsp server you are using.

However, an easier way is to create an empty file `.neoconf.json` in the develop directory

**Reply by jokajak** [reply]
I have no idea how to fix the root_dir detection so I'm going with the `.neoconf.json` solution which works magically.
